client/src/views/HomePage.vue

```diff
@@ -193,13 +194,28 @@ export default {
     Api().get('uploads/filedata/all').then((response) => {
       this.files = response.data;
     });
+    if (!this.$store.state.user) {
+      Api().get('users/self').then((response) => {
+        this.$store.dispatch('setUser', response.data);
+      }).catch((error) => {
+        console.log(error);
+      });
+    }
   },
   methods: {
+    loginWithGoogle () {
+      window.location.href = 'http://localhost:3000/auth/google';
```

Comment:
For consistency purposes I'm thinking we should reference the port environment variable here rather than redefining `3000` for the port.

Reply:
I agree, think that'll be in another PR. As it'll require proxying the requests to the server, or adding .env file in the client, or changing vite to read the .env in the server folder

Reply:
Would it be practical, and possible, to create one monolithic `.env` file for both the client and server? Thinking it might be cleaner that way.